src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -63,33 +80,160 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
                 return partition.TryGetElementAt(index, out bool _);
             }
 
+            if (source is IList<TSource> list)
```

Comment:
Likewise, we could trim down the size of this method if we integrated the checks in TryGetElement.

Reply:
Updated the code to consolidate everything into one `TryGetElementAt` method.